js/dialogcards.js

```diff
@@ -489,13 +687,13 @@ H5P.Dialogcards = (function ($, Audio, JoubelUI) {
 
       // Add next card.
       var $loadCard = self.$current.next('.h5p-dialogcards-cardwrap');
-      if (!$loadCard.length && self.$current.index() + 1 < self.params.dialogs.length) {
-        var $cardWrapper = self.createCard(self.params.dialogs[self.$current.index() + 1], self.$current.index() + 1)
+      if (!$loadCard.length && self.$current.index() + 1 < self.dialogs.length) {
+        var $cardWrapper = self.createCard(self.dialogs[self.$current.index() + 1])
```

Comment:
The second parameter seems to be missing and it's required.

Reply:
Nothing missing here. Please refer to lines 67-70 where self.params.dialogs is saved to self.dialogs for further use when save content state is used.

Reply:
Hm, seems like `createCard()` takes three arguments, but only two are required:
```
C.prototype.createCard = function (card, cardNumber, setCardSizeCallback) {
```
This call only has one, and the call on _line 439_ appear to be incorrect as well.

Reply:
You're right, I'll look into it.

Reply:
OK, fixed. Because ALL cards are loaded at once, in function C.prototype.nextCard we can skip the Add next card routine.